shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -77,52 +185,272 @@ - (BOOL)canBecomeFirstResponder {
   return YES;
 }
 
+#pragma mark - UITextInput Overrides
+
+- (id<UITextInputTokenizer>)tokenizer {
+  if (_tokenizer == nil) {
+    _tokenizer = [[UITextInputStringTokenizer alloc] initWithTextInput:self];
+  }
+  return _tokenizer;
+}
+
+- (UITextRange*)selectedTextRange {
+  return [[_selectedTextRange retain] autorelease];
+}
+
+- (void)setSelectedTextRange:(UITextRange*)selectedTextRange {
+  if (_selectedTextRange != selectedTextRange) {
+    UITextRange* oldSelectedRange = _selectedTextRange;
```

Comment:
IIRC this is Apple's recommended paranoid setter implementation. If `_selectedTextRange` has a ref-count of 1, then `release` will trigger `dealloc`, which will dealloc its start and end positions. 

If UIKit re-uses one or both positions from the current `FlutterTextRange` in the incoming selectedTextRange, then releasing the old one first would release the positions on the range that we're about to copy.